src/util.c

```diff
@@ -58,6 +59,17 @@ void util_cleanup_64(uint8_t** buf)
     util_zero(*buf, 64);
 }
 
+char* util_strdup(const char* str)
+{
+    size_t str_size = strlen(str);
+    char* result = malloc(str_size + 1);
```

Comment:
why not use `strdup` here?

Reply:
Something something because portability (nah, it's because I'm stupid. Will fix).

Reply:
Fixed! 3278d241